### ISSUE TITLE
using [flex] attribute results in class="flex-flex" which doesn't exist

### ISSUE THREAD

**Issue by mattiLeBlanc** [angular/material#5044]
Hi,

I noticed my layout went awry after updating to AM 0.11.2

The problem as I see it is that
```
<div ng-controller="AppCtrl" layout="column" flex="flex"></div>
```
turns into
```
<div ng-controller="AppCtrl" class="flex-flex md-layout-column"></div>
```
This `flex-flex` class doesn't exist and should be `flex'.

Am I correct to suggest that this is a pretty big bug?

To fix my layout I have to manually add `class="flex"` to each element to make sure it has the `flex' class

update: I do use Jade however, which compiles to `flex="flex"` which is valid

**Comment by ThomasBurleson** [comment]
`flex="flex"` is valid ? 
*  Where is this defined as a standard?
*  What style values does this equate to ?

Here are the accepted ngMaterial Flex options:  https://gist.github.com/ThomasBurleson/88152ec57c9133dec57a

**Comment by mattiLeBlanc** [comment]
I see what you mean. Apparently the Meteor JADE package converts it from `<div flex/ >` to `<div flex="flex">`.
However, before the upgrade where I was using:
angular:angular@1.4.3
angular:angular-animate@1.4.3
angular:angular-aria@1.4.3
angular:angular-material@0.10.1
angular:angular-messages@1.4.3
angular:angular-sanitize@1.4.3
angularui:angular-ui-router@0.2.15

this flex=flex didn't cause an issue.

Now after update (meteor 1.2 ) and :
angular@1.0.1
angular:angular@1.4.7
angular:angular-animate@1.4.7
angular:angular-aria@1.4.7
angular:angular-material@0.11.2
angular:angular-messages@1.4.7
angularui:angular-ui-router@0.2.15

I notice that  `<div flex/ >` becomes  `<div class="flex-flex" / >`

I checked my other project that is still running the older packages and their the JADE package also creates a `flex="flex"` attribute.

I can see if I can make a pull request with a change that it should level empty attributes alone.
What are your thoughts?

**Comment by ThomasBurleson** [comment]
We will update Layouts to safeguard against `flex="flex"`

**Comment by mattiLeBlanc** [comment]
awesome. And I will look into fixing the Jade package so that it doesn't create `flex="flex"` necessarily.
Great work.

**Comment by clshortfuse** [comment]
@mattiLeBlanc  I had the same issue, but the right way to do it is: 

    div(flex='')

**Comment by mattiLeBlanc** [comment]
I just base it on the docs of Angular Material and Polymer Elements. They all use layout and flex as attribute possibly without value.
I think it is loosely allowed, same with things like 'checked' or 'selected' in HTML

**Comment by ThomasBurleson** [comment]
Closed with SHA 2478f1d2b064d5e0
